kernel/kernel-api/src/main/java/io/delta/kernel/statistics/DataFileStatistics.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.statistics;
+
+import static java.time.ZoneOffset.UTC;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.types.*;
+import io.delta.kernel.utils.JsonUtil;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+/** Statistics about data file in a Delta Lake table. */
+public class DataFileStatistics {
+  private StructType dataSchema;
+  private final long numRecords;
+  private final Map<Column, Literal> minValues;
+  private final Map<Column, Literal> maxValues;
+  private final Map<Column, Long> nullCounts;
+
+  public static final int MICROSECONDS_PER_SECOND = 1_000_000;
+  public static final int NANOSECONDS_PER_MICROSECOND = 1_000;
+
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX");
+
+  /**
+   * Create a new instance of {@link DataFileStatistics}.
+   *
+   * @param dataSchema Schema of the data file.
+   * @param numRecords Number of records in the data file.
+   * @param minValues Map of column to minimum value of it in the data file. If the data file has
+   *     all nulls for the column, the value will be null or not present in the map.
+   * @param maxValues Map of column to maximum value of it in the data file. If the data file has
+   *     all nulls for the column, the value will be null or not present in the map.
+   * @param nullCounts Map of column to number of nulls in the data file.
+   */
+  public DataFileStatistics(
```

Comment:
I think at a high-level we should define what we expect connectors to input here, and what rules we expect them to follow in order to write correct and useful statistics. And then we should enforce them :) We should treat this constructor as a public API as connectors will need to create these. I think I left general comments all over to this nature but this is kind of the gist of it.

Reply:
one thing to emphasize in the docs is the stats are for the data written in the file and column names here are the same as the ones used in the Parquet file.

Later on when we move to id based stats, we can update it.